### PR TITLE
FE-1225: Remove the custom useRouter hook in favor of useLocation, useHistory and usePageFilters

### PR DESCRIPTION
**Pull request by shrirupa**
FE-1225: Remove the custom useRouter hook in favor of useLocation, useHistory and usePageFilters

### What Changed

- Remove and Replace the custom useRouter hook with usePageFilters, useLocation and useHistory
- Move the updateRoute/qs logic in usePageFilters
- Add storybook documentation for usePageFilters

### How To Test

- Looking for feedback on usePageFilters storybook description

- Replacement of useRouter hasn't caused any bugs in 
- [ ] ChangePlanForm
- [ ] Blocks/IncidentsPage
- [ ] Domains/CreateForm
- [ ] inbox-placement/TestListPage
- [ ] report-builder
- [ ] ScheduledReportCreatePage
- [ ] AddFilterLink
- [ ] ReportOptions
- [ ] SaveReportModal
- [ ] HealthScoreDashboard

### To Do
- [ ] Address feedback

Note - There are three unit tests that I have commented out, looking for feedback on them too

### CODE REVIEW

File: src/pages/reportBuilder/components/SavedReportsSection/tests/SaveReportModal.test.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 import { shallow } from 'enzyme';
 import { SaveReportModal } from '../SaveReportModal';
-import useRouter from 'src/hooks/useRouter';
-jest.mock('src/hooks/useRouter');
-useRouter.mockReturnValue({
-  location: { search: '?metrics=sent_count' },
-});
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
```
Comment (nicklemmon) [inline src/pages/reportBuilder/components/SavedReportsSection/tests/SaveReportModal.test.js:6]: I'm surprised you have to do this?

File: src/pages/signals/tests/IntegrationPage.test.js [hunk 2]
```diff
@@ -134,31 +159,40 @@ describe('IntegrationPage', () => {
...
+  //   wrapper.update();
+
+  //   expect(wrapper.find('CursorPaging')).toHaveProp('currentPage', 2);
+  //   expect(getIngestBatchEvents).toHaveBeenCalledWith({
+  //     batchIds: [],
+  //     cursor: 'ABCDEFC',
+  //     perPage: 10,
+  //     statuses: undefined,
+  //   });
+  // });
```
Comment (nicklemmon) [inline src/pages/signals/tests/IntegrationPage.test.js:185]: Did you mean to re-introduce this?

File: src/pages/templates/hooks/useEditorNavigation.js [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+
 const useEditorNavigation = () => {
-  const { history, requestParams: { id, version = 'draft', navKey = '', subaccount: subaccountId }} = useRouter();
-  const setNavigation = (nextNavigationKey) => {
-    history.push(`/${routeNamespace}/edit/${id}/${version}/${nextNavigationKey}${setSubaccountQuery(subaccountId)}`);
+  const history = useHistory();
+  const {
+    filters: { id, version = 'draft', navKey = '', subaccount: subaccountId },
+  } = usePageFilters(initFilters);
+  const [navKeyTemp, setNavKeyTemp] = useState(navKey.toLowerCase());
```
Comment (nicklemmon) [inline src/pages/templates/hooks/useEditorNavigation.js:20]: This is probably a better way to do it, but does the old way not work?

**Review by nicklemmon — CHANGES_REQUESTED** [review]
@shrirupa A little bit of feedback - I'm going to do some additional testing as well, but these items you can at least take a look at in the meantime.

File: src/pages/reportBuilder/components/SavedReportsSection/tests/SaveReportModal.test.js (hunk 1 above)
Comment (shrirupa) [reply src/pages/reportBuilder/components/SavedReportsSection/tests/SaveReportModal.test.js:6]: Good point! I was able to remove it from most of the places but still couldn't get the useEditorNavigation.test.js working without `jest.requireActual('react-router-dom'),`

File: src/pages/signals/tests/IntegrationPage.test.js (hunk 2 above)
Comment (shrirupa) [reply src/pages/signals/tests/IntegrationPage.test.js:185]: yeah I am still not sure how to get these two unit tests in this file working  🤔

Comment (shrirupa) [reply src/pages/signals/tests/IntegrationPage.test.js:185]: I am thinking of adding cypress test instead for this page

**Review by nicklemmon — CHANGES_REQUESTED** [review]
@shrirupa Had a few comments, re: the new Cypress tests. The features impacted appear to be working still

File: cypress/tests/integration/configuration/integrationsPage.spec.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  it('on clicking next page, api is called with correct params', () => {
+    cy.stubRequest({
+      url: `api/v1/events/ingest*`,
+      fixture: 'events/ingest/200.get.json',
+      requestAlias: 'getIngestFirst',
+      statusCode: 200,
+    });
+    cy.visit(PAGE_BASE_URL);
+    cy.wait(['@getIngestFirst', '@accountReq']);
+    cy.findByDataId('next-page').click();
```
[the remark this answers is not in the document]
Comment (shrirupa) [reply cypress/tests/integration/configuration/integrationsPage.spec.js:71]: yeah, this one looks like an ally bug

File: src/components/matchbox/Pager.js [hunk 5]
```diff
@@ -20,21 +20,21 @@ function Next(props) {
...
   const [state] = useHibana();
   const { isHibanaEnabled } = state;
 
   if (!isHibanaEnabled) {
-    return <OGPager.Previous {...omitSystemProps(props)} />;
+    return <OGPager.Previous {...omitSystemProps(props)} data-id="previous-page" />;
   }
 
-  return <HibanaPager.Previous {...props} />;
+  return <HibanaPager.Previous {...props} data-id="previous-page" />;
```
[the remark this answers is not in the document]
Comment (shrirupa) [reply src/components/matchbox/Pager.js:37]: right, this looks like an ally bug, but I think till we get this fixed, can we have data-id's here for these test ?

**Review by shrirupa — COMMENTED** [review]
@nicklemmon this is ready for final pass, whenever you have some time.

File: cypress/tests/integration/configuration/integrationsPage.spec.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      url: `api/v1/events/ingest*`,
+      fixture: 'events/ingest/200.get.json',
+      requestAlias: 'getIngestFirst',
+      statusCode: 200,
+    });
+    cy.visit(PAGE_BASE_URL);
+    cy.wait(['@getIngestFirst', '@accountReq']);
+    cy.url().should('include', PAGE_BASE_URL);
+    cy.title().should('include', 'Signals | Integration');
+    cy.findByText('Review the health of your Signals integration.').should('be.visible');
```
[the remark this answers is not in the document]
Comment (shrirupa) [reply cypress/tests/integration/configuration/integrationsPage.spec.js:25]: looks like this is ally bug bug too. PageDescription doesn't have role of heading

File: src/components/matchbox/Pager.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicklemmon) [reply src/components/matchbox/Pager.js:37]: I looked at the components and they do have content within them:
```html
<button type="button" class="sc-bdVaJa sc-jTzLTM bwdivw">
  <span class="sc-jzJRlG hgvSdV">
    <svg></svg>
    <span class="sc-htoDjs hzTsVD">Next</span>
  </span>
</button>
```

You can't grab them with `cy.findByRole('button', { name: 'Next' })`?

File: cypress/tests/integration/configuration/integrationsPage.spec.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (nicklemmon) [reply cypress/tests/integration/configuration/integrationsPage.spec.js:25]: Nope, you're right actually - `PageDescription` is just a paragraph, which oddly enough does not have a relevant `role`. My feedback was wrong!

File: src/components/matchbox/Pager.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (shrirupa) [reply src/components/matchbox/Pager.js]: Thanks for fixing this!

**Review by nicklemmon — APPROVED** [review]
@shrirupa Looks good to me!

**Pull request merged**